24429 keychains creation page

d-kochnov-optimus opened the pull request.

d-kochnov-optimus committed:
Refs #24502: prepare env for artworks uploader

d-kochnov-optimus committed:
Merge remote-tracking branch 'origin/24251-petsies-demo' into 24429-keychains-creation-page

d-kochnov-optimus committed:
Refs #24503: add budsies module

d-kochnov-optimus committed:
Refs #24503: add budsies module

gorbunovav committed:
Refs #24503: tsconfig.json - use strict mode to make type checks work

d-kochnov-optimus committed:
Refs #24525: add action for extra faces retrieving

d-kochnov-optimus committed:
Refs #24525: allow force load
extend product variant retrieve on bundle products
add cart id into add to cart action

gorbunovav committed:
Merge branch '24429-keychains-creation-page' of https://github.com/BudsiesApp/vue-storefront into 24429-keychains-creation-page

d-kochnov-optimus committed:
Refs #24606: update cart sync process to respect custom products

d-kochnov-optimus committed:
Refs #24606: keep custom product image after qty change in cart

d-kochnov-optimus committed:
Refs #24606: fix addons getter

gorbunovav commented on core/modules/cart/types/CartItem.ts:
```diff
@@ -12,5 +12,8 @@ export default interface CartItem extends Product {
+  quoteId?: string,
+  uploadedArtworkIds?: string[],
+  plushieId?: number,
+  plushieThumb?: string
```
можем немного поменять названия для большей универсальности?
* uploadedArtworkIds -> customerImagesIds
* plushieThumb -> thumbnail

gorbunovav commented on core/modules/cart/types/Servertem.ts:
```diff
@@ -0,0 +1,13 @@
+  type_id: any,
+  plushieId?: string,
+  plushieThumb?: string,
+  uploadedArtworkIds?: string[]
```
и тут тоже

gorbunovav commented on src/modules/budsies/models/addon.model.ts:
```diff
@@ -0,0 +1,25 @@
+  public id;
+
+  private label: string;
+  private value: number;
```
тут видимо их public нужно сделать

gorbunovav commented on src/modules/budsies/store/actions.ts:
```diff
@@ -0,0 +1,68 @@
+}
+
+export const actions: ActionTree<BudsiesState, RootState> = {
+  async addPrintedProductToCart (
```
а этот экшен нам еще нужен?

gorbunovav committed:
Merge branch '24251-petsies-demo' into 24429-keychains-creation-page

gorbunovav committed:
Merge branch '24251-petsies-demo' into 24429-keychains-creation-page

d-kochnov-optimus committed:
Refs #24630:
rename custom cart and server item fields
fix addon model fields access
remove unused action

d-kochnov-optimus committed:
Refs #24630: fix mutations to be reactive
add backend products enum

gorbunovav committed:
Refs #24630: Move file processing repository into the main app repository

gorbunovav committed:
Refs #24630: Budsies module - refactor factories, models, repositories, add type guards for api responses

gorbunovav committed:
Refs #24630: Move file storage-related files into a separate module

gorbunovav committed:
Refs #24630: Tweak ESLint rules
